griptape/drivers/sql/snowflake_sql_driver.py

```diff
@@ -0,0 +1,85 @@
+from typing import Callable, Optional
+from griptape.drivers import BaseSqlDriver
+from attr import Factory, define, field
+from sqlalchemy import create_engine, text, MetaData, Table
+from sqlalchemy.engine import Engine
+from sqlalchemy.exc import NoSuchTableError
+from snowflake.connector import SnowflakeConnection
+
+
+@define
+class SnowflakeSqlDriver(BaseSqlDriver):
+    snowflake_connection_function: Callable[[], SnowflakeConnection] = field(
```

Comment:
This feels a bit long. Should we shorten it? May be `connection_func`?

Reply:
I like that, yes I'll shorten it in the next commit